src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -431,7 +431,7 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
 
                 Vector128<float> scalarVector128 = Sse.SetAllVector128(scalar);
 
-                if (pDstCurrent + 4 <= pDstEnd)
+                if (pDstCurrent <= pDstEnd - 4)
```

Comment:
creating a temp for `pDstEnd - 4` is probably better, as that would better ensure that each loop isn't doing `pDstEnd - 4` before the comparison